### ISSUE TITLE
force:apex:test:run CLIError: timed out

### ISSUE THREAD

**Issue by pawel-id** [forcedotcom/cli#1909]
### Summary

Running long apex tests via `sfdx force:apex:test:run` throws error 'CLIError: timed out'

### Steps To Reproduce:

The issue is happening only on large projects with large amount of apex test. Our apex tests normally last around 6+ hours to process.

Every time after 3-5 hours duration the tests fails. It prints partial summary. Then the output is cut and then error trace from sfdx. Sfdx return 0 exit code. 

See example:

```
$ sfdx force:apex:test:run -c -v -r junit -w 480  -l RunLocalTests -d tests/apex
WARNING: WARNING: In the Summer ’21 release, we'll rename or remove some of the output files that are created when you run force:apex:test:run or force:apex:test:report with the --outputdir (-d) parameter.  This breaking change could impact your CI/CD scripts if they reference specific filenames. See https://sfdc.co/apex-test-updates for details

<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
    <testsuite name="force.apex" timestamp="2023-02-01T02:10:44.000Z" hostname="https://orangepolska--int1.sandbox.my.salesforce.com" tests="1579" failures="0"  errors="0"  time="2118.07">
        <properties>
            <property name="outcome" value="Successful"/>
            <property name="testsRan" value="1579"/>
            <property name="passing" value="1579"/>
            <property name="failing" value="0"/>
            <property name="skipped" value="0"/>
            <property name="passRate" value="100%"/>
            <property name="failRate" value="0%"/>
            <property name="testStartTime" value="Wed Feb 01 2023 2:10:44 AM"/>
            <property name="testExecutionTime" value="2118.07 s"/>
            <property name="testTotalTime" value="2118.07 s"/>
            <property name="commandTime" value="0.70 s"/>
            <property name="hostname" value="https://orangepolska--int1.sandbox.my.salesforce.com"/>
            <property name="orgId" value="00D7Y0000001Vd3UAE"/>
            <property name="username" value="cicd@orange.com.navio.int1"/>
            <property name="testRunId" value="7077Y00003kPzNW"/>
            <property name="userId" value="005090000027sksAAA"/>
            <property name="coveredLines" value="21708"/>
            <property name="testRunCoverage" value="84%"/>
            <property name="orgWideCoverage" value="79%"/>
        </properties>
        <testcase name="testUpdateAcceptanceStatusOnOrderCart" classname="Marketing_CompetenceMapStatusRATest" time="11.17">
        <!-- ... readacted a lot of regular lines ... -->
        <testcase name="testGetProductsForOffer" classname="ProductHierarchyHelperTest" time="0.05">
        </testcase>
        <testcase name="testProductOrderEventCreatePositive"CLIError: timed out
    at Object.error (/usr/local/lib/node_modules/sfdx-cli/node_modules/@oclif/core/lib/errors/index.js:28:15)
    at /usr/local/lib/node_modules/sfdx-cli/node_modules/@oclif/core/lib/cli-ux/index.js:25:66
    at async flush (/usr/local/lib/node_modules/sfdx-cli/node_modules/@oclif/core/lib/cli-ux/index.js:124:9) {
  oclif: { exit: 2 },
  code: undefined
}
ERROR running force:apex:test:run:  Unknown error in Apex Library: timed out
```

### System Information

- linux bash shell
- sfdx-cli/7.185.0 linux-x64 node-v18.13.0

**Comment by sanderott** [comment]
As of the February 1st we consistently have the same problem on two of our steps;

1. Oddly enough while updating tracked source data after a push has already succeeded. (running `sfdx force:source:push -u $ORG_ALIAS -f --wait 60`)

```
14:14:24  Pushing v56.0 metadata to test-e1xmoxzc7nro@example.com using the v57.0 REST API
14:14:31  DEPLOY PROGRESS | ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 0/3854 Components
<!-- (...) -->
14:33:19  DEPLOY PROGRESS | ████████████████████████████████████████ | 3871/3871 Components
14:33:37  Updating source tracking... done
14:33:37  === Pushed Source
14:33:37   STATE   FULL NAME                                                                        TYPE                        PROJECT PATH
<!-- (...) -->
14:33:37   Created KaWeFisStatusBuilder                                                             ApexClass                   kwf-app/kawefis/classes/test/KaWeFisStatusBuilder.cls
14:33:37  CLIError: timed out
14:33:37      at Object.error (/opt/buildagent/work/e6ed28d6ed3951dd/sfdx2/node_modules/@oclif/core/lib/errors/index.js:28:15)
14:33:37      at /opt/buildagent/work/e6ed28d6ed3951dd/sfdx2/node_modules/@oclif/core/lib/cli-ux/index.js:25:66
14:33:37      at async flush (/opt/buildagent/work/e6ed28d6ed3951dd/sfdx2/node_modules/@oclif/core/lib/cli-ux/index.js:124:9) {
14:33:37    oclif: { exit: 2 },
14:33:37    code: undefined
14:33:37  }
14:33:37   Created KaWeFisStatusBuilder                                                             ApexClass                   kwf-app/kawefis/classes/test/KaWeFisStatusBuilder.cls-meta.xml
<!-- (...) -->
```

2. As mentioned in the original summary, when running `sfdx force:apex:test:run`

```
14:46:58   <?xml version="1.0" encoding="UTF-8"?>
14:46:58   <testsuites>
14:46:58       <testsuite name="force.apex" timestamp="2023-02-02T13:35:27.000Z" hostname="https://rockyroad-pisces-7910.scratch.my.salesforce.com" tests="1155" failures="4"  errors="0"  time="1315.94">
14:46:58           <properties>
14:46:58               <property name="outcome" value="Failed"/>
14:46:58               <property name="testsRan" value="1155"/>
14:46:58               <property name="passing" value="1151"/>
14:46:58               <property name="failing" value="4"/>
14:46:58               <property name="skipped" value="0"/>
14:46:58               <property name="passRate" value="100%"/>
14:46:58               <property name="failRate" value="0%"/>
14:46:58               <property name="testStartTime" value="Thu Feb 02 2023 2:35:27 PM"/>
14:46:58               <property name="testExecutionTime" value="1315.94 s"/>
14:46:58               <property name="testTotalTime" value="1315.94 s"/>
14:46:58               <property name="commandTime" value="0.59 s"/>
14:46:58               <property name="hostname" value="https://rockyroad-pisces-7910.scratch.my.salesforce.com"/>
14:46:58               <property name="orgId" value="00D8D0000006jlhUAA"/>
14:46:58               <property name="username" value="test-e1xmoxzc7nro@example.com"/>
14:46:58               <property name="testRunId" value="7078D00001Y7qNc"/>
14:46:58               <property name="userId" value="0058D000003gGCqQAM"/>
14:46:58               <property name="coveredLines" value="9649"/>
14:46:58               <property name="testRunCoverage" value="93%"/>
14:46:58               <property name="orgWideCoverage" value="92%"/>
14:46:58           </properties>
14:46:58           <testcase name="buildKey_withoutNamespace_shouldReturnError" classname="CacheManagerTest" time="0.02">
<!-- (...) -->
14:46:58          <testcase name="testIsUsernamePasswordEnabled" classname="LightningLoginFormControllerTest" time="0.03">
14:46:58  CLIError: timed out
14:46:58      at Object.error (/opt/buildagent/work/e6ed28d6ed3951dd/sfdx2/node_modules/@oclif/core/lib/errors/index.js:28:15)
14:46:58      at /opt/buildagent/work/e6ed28d6ed3951dd/sfdx2/node_modules/@oclif/core/lib/cli-ux/index.js:25:66
14:46:58      at async flush (/opt/buildagent/work/e6ed28d6ed3951dd/sfdx2/node_modules/@oclif/core/lib/cli-ux/index.js:124:9) {
14:46:58    oclif: { exit: 2 },
14:46:58    code: undefined
14:46:58  }
14:46:58          </testcase>
14:46:58          <testcase name="testLoginWithInvalidCredentials" classname="LightningLoginFormControllerTest" time="0.01">
<!-- (...) -->
14:46:58          </testcase>
14:46:58  ERROR running force:apex:test:run:  Unknown error in Apex Library: timed out
14:46:58          <testcase name="testGetNullExtraFields" classname="LightningSelfRegisterControllerTest" time="0.01">
<!-- (...) -->
```

### System Information

```
sfdx-cli/7.185.0 linux-x64 node-v18.12.1
   {
     "cliVersion": "sfdx-cli/7.185.0",
     "architecture": "linux-x64",
     "nodeVersion": "node-v18.12.1",
     "pluginVersions": [
       "@oclif/plugin-autocomplete 1.3.10 (core)",
       "@oclif/plugin-commands 2.2.2 (core)",
       "@oclif/plugin-help 5.1.22 (core)",
       "@oclif/plugin-not-found 2.3.13 (core)",
       "@oclif/plugin-plugins 2.1.8 (core)",
       "@oclif/plugin-search 0.0.8 (core)",
       "@oclif/plugin-update 3.0.9 (core)",
       "@oclif/plugin-version 1.1.4 (core)",
       "@oclif/plugin-warn-if-update-available 2.0.19 (core)",
       "@oclif/plugin-which 2.2.6 (core)",
       "alias 2.1.18 (core)",
       "apex 1.4.2 (core)",
       "auth 2.3.12 (core)",
       "community 2.1.3 (core)",
       "config 1.4.23 (core)",
       "custom-metadata 2.0.11 (core)",
       "data 2.1.22 (core)",
       "generator 2.0.16 (core)",
       "info 2.3.3 (core)",
       "limits 2.2.3 (core)",
       "org 2.2.22 (core)",
       "packaging 1.12.3 (core)",
       "schema 2.2.2 (core)",
       "signups 1.2.12 (core)",
       "source 2.3.15 (core)",
       "telemetry 2.0.5 (core)",
       "templates 55.2.1 (core)",
       "trust 2.2.8 (core)",
       "user 2.1.25 (core)",
       "@salesforce/sfdx-plugin-lwc-test 1.0.1 (core)",
       "salesforce-alm 54.8.5 (core)"
     ],
     "osVersion": "Linux 5.4.0-137-generic",
     "shell": "unknown",
     "rootPath": "/opt/buildagent/work/e6ed28d6ed3951dd/sfdx2"
   }
```

**Comment by daveespo** [comment]
Likely dupe of #1903

**Comment by shetzel** [comment]
Thank you for reporting.  Please update to the latest version of the CLI published today, v7.186.2, which has the fix for this.

**Comment by pawel-id** [comment]
Thank you. Confirm that v7.186.2 fix the issue.
